main.go

```diff
@@ -347,3 +353,7 @@ func ensureSerial(s string) (string, error) {
 
 	return s, nil
 }
+
+func timeoutToMs(t time.Duration) time.Duration {
```

Comment:
Changed the type of the timeout in the command line to uint32 (mostly to solve the compiler madness below), so I went with approach 2 by changing the function.
I prefer to avoid approach 1 because I don't think a user would need the ability to specify the timeout at that level of detail (feels like a potential footgun).
So it's approach 2, but without the "-ms" suffix in the parameter, because the fact that the value is in milliseconds is already in the description you get with --help

Reply:
Sounds good.

As you'll learn about me - I'm a fan of explicit units on otherwise ambiguous parameter / variable names - mainly because it saves me the extra second of having to look at comments or documentation 😄 But since milliseconds is the logical unit of choice for this kind of tool, I agree it's probably OK to leave the suffix off.